Backendless/Utils/DeviceCheck.cs

```diff
@@ -0,0 +1,35 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+using Weborb.Util;
+
+namespace BackendlessAPI.Utils
+{
+  internal static class DeviceCheck
+  {
+    private const String XAMARIN_FULLNAME = "Xamarin.Forms.Device, Xamarin.Forms.Core, Version=2.0.0.0, Culture=neutral, PublicKeyToken=null";
+    
+    internal static Boolean IsMobile
+    {
+      get => TypeLoader.LoadType( XAMARIN_FULLNAME ) != null;
+    }
+
+    internal static String GetDeviceOs()
+    {
+      try
+      {
+        Type currentDeviceType = TypeLoader.LoadType( XAMARIN_FULLNAME );
+
+        if( currentDeviceType != null )
+          return currentDeviceType.GetProperty( "RuntimePlatform" )
+                                                   .GetValue( currentDeviceType, null ).ToString().ToUpper();
+      }
+      catch( System.Exception e )
+      {
+        throw new ArgumentException( e.Message );
+      }
+
+      return null;
```

Comment:
not sure if the server can handle "null". Could you try to see what happens if the "Os" property is null when you register device?

Reply:
server will be return error.